src/buildApplication/getVercelConfig.ts

```diff
@@ -28,6 +28,44 @@ export async function getVercelConfig(): Promise<VercelConfig> {
 	return config;
 }
 
+/**
+ * Gets the routes of a specific vercel phase
+ *
+ * @param config Vercel config object containing all the routes information
+ * @param phase the phase from which extract the routes
+ * @returns the phases' routes
+ */
+export function getPhaseRoutes(
+	routes: VercelRoute[],
+	phase: VercelPhase
+): VercelRoute[] {
+	if (!routes.length) {
+		return [];
+	}
+
+	const phaseStart = getPhaseStart(routes, phase);
+	const phaseEnd = getPhaseEnd(routes, phaseStart);
+	return routes.slice(phaseStart, phaseEnd);
+}
+
+function getPhaseStart(routes: VercelRoute[], phase: VercelPhase): number {
+	if (phase === 'none') {
+		return 0;
+	}
+
+	const index = routes.findIndex(
+		route => isVercelHandler(route) && route.handle === phase
+	);
+	return index === -1 ? routes.length : index + 1;
+}
+
+function getPhaseEnd(routes: VercelRoute[], phaseStart: number): number {
+	const index = routes
+		.slice(phaseStart)
+		.findIndex(route => isVercelHandler(route));
```

Comment:
The slice is a bit of heavy way to skip the first few items. How about:

```suggestion
		.findIndex((route, i) => i >= phaseStart && isVercelHandler(route));
```

Reply:
sure 🙂